grading/execute.cpp

```diff
@@ -70,23 +62,23 @@ bool system_program(const std::string &program, std::string &full_path_executabl
     for (auto& entry : allowed_autograding_commands.items()) {
       nlohmann::json program_obj = entry.value();
       if (program_obj["path"] == program) {
-        if (command_available_in_env(program_obj, running_in_docker)) {
-          full_path_executable = program;
-          return true;
-        }
-        return false;
+        full_path_executable = program;
+        return true;
       }
     }
   }
   else {
     if (allowed_autograding_commands.contains(program)) {
       nlohmann::json program_obj = allowed_autograding_commands[program];
-      if (command_available_in_env(program_obj, running_in_docker)) {
-        full_path_executable = program_obj["path"];
-        return true;
-      }
+      full_path_executable = program_obj["path"];
+      return true;
     }
   }
+  // if we are docker, allow any program. previous code is necessary
+  // to run before return for full path expansion.
+  if (running_in_docker) {
+    return true;
```

Comment:
It looks like full_path_executable isn't getting set here. When I tested, this was causing validate_program to return an empty string and the assertion on line 513 to fail in execution.